server.go

```diff
@@ -274,23 +274,26 @@ func (s *baseServer) accept(ctx context.Context) (quicSession, error) {
 
 // Close the server
 func (s *baseServer) Close() error {
-	s.sessionHandler.CloseServer()
 	s.mutex.Lock()
-	defer s.mutex.Unlock()
 	if s.closed {
+		s.mutex.Unlock()
 		return nil
 	}
 	if s.serverError == nil {
 		s.serverError = errors.New("server closed")
 	}
-	var err error
 	// If the server was started with ListenAddr, we created the packet conn.
 	// We need to close it in order to make the go routine reading from that conn return.
-	if s.createdPacketConn {
-		err = s.sessionHandler.Destroy()
-	}
+	createdPacketConn := s.createdPacketConn
 	s.closed = true
 	close(s.errorChan)
+	s.mutex.Unlock()
+
+	s.sessionHandler.CloseServer()
+	var err error
+	if createdPacketConn {
+		err = s.sessionHandler.Destroy()
+	}
 	<-s.running
```

Comment:
I think `CloseServer()` could still be called before `(*baseServer).run` finishes processing a packet and calls `AddWithConnID` with the same end result.

I think receiving `<-s.running` is necessary before calling `CloseServer()` to ensure that `(*baseServer).run` is not processing a packet. 
```golang
	<-s.running
	s.sessionHandler.CloseServer()
  var err error
	if createdPacketConn {
		err = s.sessionHandler.Destroy()
	}
```

Reply:
Right. Thank you!